decentraland/screens/ListLandForSaleScreen.js

```diff
@@ -4,78 +4,60 @@ import { setLandForSaleList, selectLandToBuy } from "../redux/actions";
 import PropTypes from "prop-types";
 import LandForSaleList from "@presentational/LandForSaleList";
 import LandForSaleListItem from "@presentational/LandForSaleListItem";
+import {
+  prepareParcelForSale,
+  prepareEstateForSale,
+  addressesAreEqual,
+  showError,
+  showInfo,
+  getContracts,
+} from "./helpers";
 
-import ContractsAddresses from "@constants/ContractsAddresses";
-const { ESTATE_ADDRESS, MARKETPLACE_ADDRESS } = ContractsAddresses;
-
-import { Action } from "tasit-sdk";
-const { ERC721, Marketplace } = Action;
-const { Estate } = ERC721;
-const { Decentraland: DecentralandMarketplace } = Marketplace;
+import DecentralandUtils from "tasit-sdk/dist/helpers/DecentralandUtils";
 
 export class ListLandForSaleScreen extends React.Component {
-  estateContract = new Estate(ESTATE_ADDRESS);
-  marketplaceContract = new DecentralandMarketplace(MARKETPLACE_ADDRESS);
-
   componentDidMount = async () => {
     try {
       const { setLandForSaleList } = this.props;
-      const landForSaleList = await this._getLandForSaleList();
+      showInfo("Loading assets for sale...");
+      const landForSaleList = await this._getAssetsForSale();
       setLandForSaleList(landForSaleList);
     } catch (err) {
-      console.error(err);
+      showError(err);
     }
   };
 
-  // Note: This function is assuming that:
-  // - All estates have a sell order
-  // - The total supply of estates is small
-  // TODO: Rewrite this function when we move to testnet
-  _getLandForSaleList = async () => {
-    const orders = [];
-    const totalSupply = await this.estateContract.totalSupply();
-
-    for (let estateId = 1; estateId <= Number(totalSupply); estateId++) {
-      const order = this._getLandForSale(estateId);
-      orders.push(order);
-    }
+  _getAssetsForSale = async () => {
+    const decentralandUtils = new DecentralandUtils();
+    const { getOpenSellOrders } = decentralandUtils;
 
-    if (orders.length == 0) console.warn("There isn't any land for sale right now.");
+    const fromBlock = 0;
```

Comment:
We could probably pick a higher number than this or make it dynamic (without getting too fancy - doesn't have to be perfectly accurate)

Reply:
Seems that won't make difference. I've tried to change that before and the conclusions were:
- For development ganache must be `fromBlock` 0;
- For forked ganache, even using a closest `fromBlock` number, that call is very slow (and because of that I'm using real testnet to get open orders to test over forked testnet).
- For real chains using `Infura`/`Etherscan`, one HTTP call is made using `fromBlock` as a parameter and the response time is good.
Said that I think that `0` cover all cases well. 

An improvement that I think should be made is: Changing the `DecentralandUtils` API `getOpenSellOrders(fromBlock)` to `getAllOpenSellOrders()` or even `getAllAssetsForSale()`?

Reply:
That all makes sense and sounds good. Feel free to create an issue for that improvement. I like the name `getAllAssetsForSale()`

Reply:
Clever move to use real testnet to get open orders for forked testnet, by the way

Reply:
Extracted to: https://github.com/tasitlabs/TasitSDK/issues/288

Reply:
Thanks! Resolving